src/components/App.js

```diff
@@ -2,47 +2,108 @@ import { useEffect, useState } from 'react'
 import Recipe from './Recipe'
 import Header from './Header'
 import styled from 'styled-components/macro'
-import SearchFilter from './SearchFilter'
+import SearchBar from './SearchBar'
 import Alert from './Alert'
+import CheckboxFilter from './FilterForm'
 require('dotenv').config()
 
 export default function App() {
   const [recipes, setRecipes] = useState([])
   const [alert, setAlert] = useState('')
-  const [query, setQuery] = useState('chicken')
 
-  function handleSearch(searchQuery) {
-    setQuery(searchQuery)
-  }
+  const [query, setQuery] = useState('chicken')
+  const [caloriesRangeFrom, setCaloriesRangeFrom] = useState('')
+  const [caloriesRangeTo, setCaloriesRangeTo] = useState('')
+  const [healthLabels, setHealthLabels] = useState([])
+  const [dishTypes, setDishTypes] = useState([])
 
-  useEffect(() => {
-    getRecipes()
-  }, [query])
+  const [url, setUrl] = useState(creatUrlQuery())
 
   async function getRecipes() {
     if (query !== '') {
-      const response = await fetch(
-        `https://api.edamam.com/search?q=${query}&app_id=${process.env.REACT_APP_EDAMAM_API_ID}&app_key=${process.env.REACT_APP_EDAMAM_API_KEY}&from=0&to=30`
-      )
+      let url = creatUrlQuery()
+      const response = await fetch(url)
       const data = await response.json()
-      if (data.more) {
+
+      if (data.more && data.hits) {
         setAlert('')
+        setRecipes(data.hits.map(item => item.recipe))
       } else {
-        setAlert('Cannot find recipe with such ingredient')
+        setAlert('Cannot find such recipe')
+        setRecipes([])
       }
-      setRecipes(data.hits.map(item => item.recipe))
     } else {
       setAlert('Please fill the Search Bar')
       setRecipes([])
     }
   }
 
+  useEffect(() => {
+    setUrl(creatUrlQuery())
+  }, [query, caloriesRangeFrom, caloriesRangeTo, healthLabels, dishTypes])
+
+  useEffect(() => {
+    getRecipes()
+  }, [url])
+
+  function creatUrlQuery() {
```

Comment:
Hi, would it be possible to save this function outside this file?

Reply:
Hi,good idea!Thanks for your suggestion!